mock/cpp_mock_scenarios/src/random_scenario/random001.cpp

```diff
@@ -0,0 +1,240 @@
+    engine_(seed_gen_())
+  {
+    start();
+  }
+
+private:
+  std::shared_ptr<random001::ParamListener> param_listener_;
+  random001::Params params_;
+  std::random_device seed_gen_;
+  std::mt19937 engine_;
+  double lane_change_position = 0.0;
+  bool lane_change_requested = false;
+  void spawnRoadParkingVehicles()
+  {
+    std::normal_distribution<> normal_dist(
+      0.0, params_.random_parameters.road_parking_vehicle.s_variance);
+    const auto spawn_road_parking_vehicle =
+      [&](const auto & entity_index, const auto offset, const auto number_of_vehicles) {
+        const auto get_entity_subtype = [](const std::string & entity_type) {
+          if (entity_type == "car") {
+            return traffic_simulator_msgs::msg::EntitySubtype::CAR;
+          } else if (entity_type == "truck") {
+            return traffic_simulator_msgs::msg::EntitySubtype::TRUCK;
+          } else if (entity_type == "bus") {
+            return traffic_simulator_msgs::msg::EntitySubtype::BUS;
+          } else if (entity_type == "trailer") {
+            return traffic_simulator_msgs::msg::EntitySubtype::TRAILER;
+          }
+          return traffic_simulator_msgs::msg::EntitySubtype::CAR;
+        };
+
+        std::string entity_name = "road_parking_" + std::to_string(entity_index);
+        constexpr lanelet::Id spawn_lanelet_id = 34705;
+        api_.spawn(
+          entity_name,
+          api_.canonicalize(traffic_simulator::helper::constructLaneletPose(
+            spawn_lanelet_id,
+            static_cast<double>(entity_index) / static_cast<double>(number_of_vehicles) *
+                api_.getLaneletLength(spawn_lanelet_id) +
+              normal_dist(engine_),
+            offset, 0, 0)),
+          getVehicleParameters(
+            get_entity_subtype(params_.random_parameters.road_parking_vehicle.entity_type)));
+        api_.requestSpeedChange(entity_name, 0, true);
+      };
+    std::uniform_real_distribution<> dist(
+      params_.random_parameters.road_parking_vehicle.min_offset,
+      params_.random_parameters.road_parking_vehicle.max_offset);
+    for (int i = 0; i < params_.random_parameters.road_parking_vehicle.number_of_vehicle; i++) {
+      spawn_road_parking_vehicle(
+        i, dist(engine_), params_.random_parameters.road_parking_vehicle.number_of_vehicle);
+    }
+  }
+
+  /// Despawn parking entity before replacing parking entity.
+  void despawnRoadParkingVehicles()
+  {
+    for (int i = 0; i < params_.random_parameters.road_parking_vehicle.number_of_vehicle; i++) {
+      api_.despawn("road_parking_" + std::to_string(i));
+    }
+  }
+
+  void despawnCrossingPedestrians()
+  {
+    for (int i = 0; i < params_.random_parameters.crossing_pedestrian.number_of_pedestrian; i++) {
+      std::string entity_name = "pedestrian" + std::to_string(i);
+      if (api_.entityExists(entity_name)) {
+        api_.despawn(entity_name);
+      }
+    }
+  }
+
+  void onUpdate() override
+  {
+    [&]() {
```

Comment:
Why to use a lambda function here?
I think there is no advantage to use lambda function here than {} to limit scope.